confluent_kafka/schema_registry/json_schema.py

```diff
@@ -0,0 +1,261 @@
+            registered_schema = self._registry.lookup_schema(subject,
+                                                             self._schema)
+            self._schema_id = registered_schema.schema_id
+            self._known_subjects.add(subject)
+
+        if self.to_dict is not None:
+            value = self.to_dict(ctx, obj)
+        else:
+            value = obj
+
+        try:
+            validate(instance=value, schema=self._parsed_schema)
+        except ValidationError as ve:
+            raise SerializationError(ve.message)
+
+        with _ContextStringIO() as fo:
+            # Write the magic byte and schema ID in network byte order (big endian)
+            fo.write(struct.pack('>bI', _MAGIC_BYTE, self._schema_id))
+            # write the record to the rest of the buffer
+            json.dump(obj, fo, value)
+
+            return fo.getvalue()
+
+
+class JsonDeserializer(Deserializer):
+    """
+    JsonDeserializer decodes bytes written in the Schema Registry
+    JSON format to an object.
+
+    Args:
+        schema_registry_client (SchemaRegistryClient): Confluent Schema Registry
+            client instance.
+
+        schema_str (str): JSON schema declaration.
+
+        from_dict (callable, optional): Callable(SerializationContext, dict) -> object.
+            Converts dict to an instance of some object.
+
+    .. _Schema definition:
+        https://json-schema.org/understanding-json-schema/reference/generic.html
+
+    """
+    __slots__ = ['_schema', '_registry', 'from_dict']
+
+    def __init__(self, schema_registry_client, schema_str, from_dict=None):
+        self._registry = schema_registry_client
+        self._schema = json.loads(schema_str)
+
+        if from_dict is not None and not callable(from_dict):
+            raise ValueError("from_dict must be callable with the signature"
+                             " from_dict(SerializationContext, dict) -> object")
+
+        self.from_dict = from_dict
+
+    def __call__(self, ctx, value):
+        """
+        Deserializes Schema Registry response to JSON object literal(dict)
+
+        Args:
+            ctx (SerializationContext): Metadata pertaining to the serialization
+                operation.
+
+            value (bytes): bytes
+
+        Returns:
+            dict: Deserialized JSON
+
+        Raises:
+            SerializerError: If response payload and expected Message type
+            differ.
+
+        """
+        if len(value) <= 5:
+            raise SerializationError("message is too small to decode {}".format(value))
+
+        with _ContextStringIO(value) as payload:
+            magic, schema_id = struct.unpack('>bI', payload.read(5))
+            if magic != _MAGIC_BYTE:
+                raise SerializationError(
+                    "message does not start with magic byte")
+
+            # JSON documents are self-describing; no need to query schema
+            return json.loads(payload.read(), encoding="utf8")
```

Comment:
you're not using `from_dict`, `_schema`, or `_registry`.

Reply:
i think you should apply `from_dict`, but no need to worry about `_schema` or `_registry`. open to someone else having a better idea, not sure this is the best idea here.

Reply:
that was actually an accidental removal when fixing it up to match the latest version of the generic serdes/avro api. I'll add a test since that would have caught this. Thanks for pointing it out.